add documentation on limits and move out Temporal Web docs (#433)
temporalio/documentation: docs/system-tools/web-ui.md

sidebar_label: Web UI
---

The Temporal Web UI is used to view workflows, see what's running, and explore and debug workflow executions.

For a **video demo** of how this looks, you can [check our docs](https://docs.temporal.io/docs/java-run-your-first-app/#state-visibility).

## Deploying and Securing Temporal Web

> _This section is still being written - if you have specific questions you'd like us to answer, please search or [ask on GitHub](https://github.com/temporalio/web/)._

Author: ```suggestion
_If you have specific questions you'd like us to answer, please check Slack, Forums, or [post questions on GitHub](https://github.com/temporalio/web/)._
```

| TEMPORAL_SESSION_SECRET          | Secret used to hash the session with HMAC                                       | "ensure secret in production" |
| TEMPORAL_EXTERNAL_SCRIPTS        | Additional JavaScript tags to serve in the UI                                   |                               |
| TEMPORAL_GRPC_MAX_MESSAGE_LENGTH | gRPC max message length (bytes)                                                 | 4194304 (4mb)                 |

<details>
<summary>
Optional TLS configuration variables:
</summary>

| Variable                              | Description                                                         | Default |
| ------------------------------------- | ------------------------------------------------------------------- | ------- |
| TEMPORAL_TLS_CERT_PATH                | Certificate for the server to validate the client (web) identity    |         |
| TEMPORAL_TLS_KEY_PATH                 | Private key for secure communication with the server                |         |
| TEMPORAL_TLS_CA_PATH                  | Certificate authority (CA) certificate for the validation of server |         |

> | TEMPORAL_TLS_ENABLE_HOST_VERIFICATION | Enables verification of the server certificate                      | true    |

Reviewer: https://github.com/temporalio/temporal/releases/tag/v1.9.0

this flag was changed :)

Reviewer: tls_disable_host_verification (default false) ;)

| TEMPORAL_TLS_SERVER_NAME              | Target server that is used for TLS host verification                |         |

- To enable mutual TLS, you need to specify `TEMPORAL_TLS_KEY_PATH` and `TEMPORAL_TLS_CERT_PATH`.
- For server-side TLS you need to specify only `TEMPORAL_TLS_CA_PATH`.

> By default we will also verify your server `hostname`, matching it to `TEMPORAL_TLS_SERVER_NAME`. You can turn this off by setting `TEMPORAL_TLS_ENABLE_HOST_VERIFICATION` to `false`.

Reviewer: same here

Reviewer: tls_disable_host_verification
have to disable it by setting it to true

Author: not doing this for now bc it doesnt yet apply to temporal web cc @feedmeapples

</details>

> ### Configuring Authentication (optional)

Reviewer 2: ```suggestion
### Configuring authentication
```
This causes the url to contain "-optional".
And generally I don't think it is a good practice.
I think it is easy enough to identify this section as optional in one of the first sentences.

         client_secret: xxxxxxxxxxxxxxxxxxxxxxx
         scope: openid profile email
         audience: temporal # identifier of the audience for an issued token (optional)
         callback_base_uri: http://localhost:8088
         pass_id_token: false # adds ID token as 'authorization-extras' header with every request to server
   ```

   <details>
   <summary>
   Providing <code>config.yml</code> to Docker image
   </summary>

   If you are running Temporal Web from the docker image, you can provide your external config.yml to docker to override the internal config.
   Create config.yml file on your machine, for example at `~/Desktop/config.yml`.
   Start the docker image, providing the path to your config.yml file using external volume flag (-v). Leave the path after the semicolon as is:

   ```bash
   docker run --network host -v ~/Desktop/config.yml:/usr/app/server/config.yml temporalio/web:latest
   ```

   </details>

   In future, multiple Oauth providers may be supported, however for now we only read the first Oauth provider under the `providers` key above.

   Common Oauth Providers and their docs:

>    - Google: https://developers.google.com/identity/protocols/oauth2/openid-connect

Reviewer 2: I think these all have bad indents.

   - Okta: https://help.okta.com/en/prod/Content/Topics/Apps/Apps_App_Integration_Wizard_OIDC.htm
       <details>
         <summary>
           Troubleshooting note for Okta users:
         </summary>
         Some providers like Okta, have a race condition that may cause logins to occasionally fail. You can get around this by providing the full URL to the `openid-configuration` path as part of the `issuer` parameter:

     ```yaml
       auth:
         enabled: true
         providers:
             - label: 'okta dev'
               type: oidc
               issuer: https://dev-xxxxxxx.okta.com/.well-known/openid-configuration
               ...
     ```

     </details>

   - Keycloak: https://www.keycloak.org/getting-started/getting-started-docker
   - please feel free to [PR or request more help on the Temporal Web repo](https://github.com/temporalio/web/)

2. You will need to provide a redirect URL to your Oauth Provider. If you are hosting Temporal Web at `http://localhost:8088` (this is configured by `callback_base_uri` in `server/config.yml`), then it is `http://localhost:8088/auth/sso_callback`.

>    - By default, Temporal Web asks for 3 scopes, make sure your provider recognizes these or you may see scope-related errors:

Author: ```suggestion
     - By default, Temporal Web asks for 3 scopes, make sure your provider recognizes these or you may see scope-related errors:
```

     - `openid` required by some OIDC providers like [auth0](https://auth0.com/docs/scopes/openid-connect-scopes)
     - `profile` for name
     - `email` for email

> ## Using Temporal Web for Development

Reviewer 2: ```suggestion
## Using Temporal Web for development
```

Once you have the Temporal Server running locally (use the [quick install guide](/docs/server/quick-install)), you can view the Temporal Web UI at [http://localhost:8088](http://localhost:8088) (this is configurable with the `TEMPORAL_WEB_PORT` environment variable).

> > ⚠️ This is a basic guide to troubleshooting/debugging Temporal applications. It is work-in-progress and we encourage [reading about our Architecture](https://docs.temporal.io/docs/server-architecture) for more detail. The better you understand how Temporal works, the better you will be at debugging your workflows.

Reviewer 2: ```suggestion
> ⚠️ This is a basic guide to troubleshooting/debugging Temporal applications.
> It is work-in-progress and we encourage [reading about our Architecture](https://docs.temporal.io/docs/server-architecture) for more detail.
The better you understand how Temporal works, the better you will be at debugging Workflow Executions.
```
